clickhouse/client.h

```diff
@@ -147,16 +146,45 @@ struct ClientOptions {
         */
         DECLARE_FIELD(context_options, int, SetContextOptions, DEFAULT_VALUE);
 
-        /** Use SNI at ClientHello and verify that certificate is issued to the hostname we are trying to connect to
+        /** Use SNI at ClientHello
          */
         DECLARE_FIELD(use_sni, bool, SetUseSNI, true);
 
+        /** Skip SSL session verification (server's certificate, etc).
+         *
+         *  WARNING: settig to true will bypass all SSL session checks, which
+         *  is dangerous, but can be used against self-signed certificates, e.g. for testing purposes.
+         */
+        DECLARE_FIELD(skip_verification, bool, SetSkipVerification, false);
+
+        /** Mode of verifying host ssl certificate against name of the host, set with SSL_set_hostflags.
+         *  For details see https://www.openssl.org/docs/man1.1.1/man3/SSL_set_hostflags.html
+         */
+        DECLARE_FIELD(host_flags, int, SetHostVerifyFlags, DEFAULT_VALUE);
```

Comment:
Test? For example maybe we can trigger  X509_CHECK_FLAG_NO_WILDCARDS? I guess it should make the `github.demo.trial.altinity.cloud` stop working, because it uses wildcard cert:
```
depth=0 CN = *.demo.trial.altinity.cloud
```

Reply:
Rn, it is too hard to set up any complex tests with the tools we have

But I already asked QA team to do the in-depth tests with various tricky ssl (mis)configurations. They'll do it once they have resources.